### ISSUE TITLE
AMD support

### ISSUE THREAD

**Issue by arboleya** [thinkpixellab/PxLoader#25]
Even tough there seems to be some work done for providing AMD compatibility, there are some inconsistencies that may still fail.

For example, I can have:

````coffeescript
define [
  'pxloader'
  'pxloaderimage'
  'pxloadersound',
  'pxloadervideo'
], ( pxloader, pxloadersound, pxloaderimage,  pxloadervideo)->
    # code here...
````

Which may fail, since `PxLoaderimage`, `PxLoaderSound` and `PxLoaderVideo` will try to add some methods on `PxLoader` prototype through it's global definition.

````javascript
PxLoader.prototype.addVideo = function(url, tags, priority) {
PxLoader.prototype.addSound = function(id, url, tags, priority) {
PxLoader.prototype.addImage = function(url, tags, priority) {
````

As AMD **do not** load files orderly, the `sound`, `image` and `video` plugins may finish loading first, specially because they are much smaller. And every time it happens, an error will be raised.

````javascript
Uncaught ReferenceError: PxLoader is not defined 
````

In order to fix this, this raw prototype calls should be made as a AMD `require` as well in case `define` and `amd.define` is present, so the `PxLoader` reference will never be `undefined`.

**Comment by robinnorth** [comment]
This is fixed by #48 😄

**Comment by robinnorth** [comment]
@joelfillmore: This can be closed now :)
